framework/web/Session.php

```diff
@@ -129,6 +131,7 @@ public function open()
         if ($this->getIsActive()) {
             Yii::info('Session started', __METHOD__);
             $this->updateFlashCounters();
+            register_shutdown_function([$this, 'close']);
```

Comment:
Yes it is OK. Such approach usually used if you are dealing with standard PHP session based on files to avoid blocking for the cuncurrent web requests.